src/source/source_state.js

```diff
@@ -130,9 +128,16 @@ class SourceFeatureState {
                 this.state[sourceLayer]  = this.state[sourceLayer] || {};
                 const layerStates = {};
                 for (const feature in this.deletedStates[sourceLayer]) {
-                    for (const key in this.deletedStates[sourceLayer][feature]) {
-                        if (this.state[sourceLayer][feature][key]) this.state[sourceLayer][feature][key] = null;
+
+                    const featureProperties = Object.keys(this.deletedStates[sourceLayer][feature]);
```

Comment:
`Object.keys` will throw when `this.deletedStates[sourceLayer][feature] == null`. Needs a null check instead of the `length == 0` test in ln133-134

Reply:
👍 not caught by tests -- only surfaces when the remove happens on a separate coalesce from the original `set`